src/IR/types.ts

```diff
@@ -6,56 +6,89 @@ export interface IntegerType {
   low?: bigint;
   high?: bigint;
 }
+export interface TextType {
+  type: "text";
+  capacity?: number;
```

Comment:
`Infinity` could make more sense here instead of `undefined` to convey "unbounded." Also avoids extra checks in the operations involving it.

Alternatively, this size could be an IntegerType with the existing notations of unbounded intervals, but there's no need for the minimum length.

Reply:
Or capacity could be a bigint to keep with our convention of internals using bigints.

Reply:
> Or capacity could be a bigint to keep with our convention of internals using bigints.

Thought about that, but array length is not `bigint` either.